status/health/window/key_error_source_test.go

```diff
@@ -544,3 +544,125 @@ func TestAnchoredMultiKeyHealthyIfNotAllErrorsSource_GapThenRepairingThenError(t
 		},
 	}, source.HealthStatus(ctx))
 }
+
+func TestMultiKeyUnhealthyIfNoRecentErrorsSource(t *testing.T) {
+	messageInCaseOfError := "message in case of error"
+	for _, testCase := range []struct {
+		name          string
+		keyErrorPairs []keyErrorPair
+		expectedCheck health.HealthCheckResult
+		timeProvider  TimeProvider
+	}{
+		{
+			name:          "healthy when there are no items",
+			keyErrorPairs: nil,
+			expectedCheck: whealth.HealthyHealthCheckResult(testCheckType),
+			timeProvider:  NewOrdinaryTimeProvider(),
+		},
+		{
+			name: "healthy when all keys are completely healthy",
+			keyErrorPairs: []keyErrorPair{
+				{key: "1"},
+				{key: "1"},
+				{key: "2"},
+				{key: "3"},
+			},
+			expectedCheck: whealth.HealthyHealthCheckResult(testCheckType),
+			timeProvider:  NewOrdinaryTimeProvider(),
+		},
+		{
+			name: "unhealthy when some keys are unhealthy",
+			keyErrorPairs: []keyErrorPair{
+				{key: "1"},
+				{key: "1", err: werror.Error("Error #1 for key 1")},
+				{key: "1"},
+				{key: "2", err: werror.Error("Error #1 for key 2")},
+				{key: "3"},
+			},
+			expectedCheck: health.HealthCheckResult{
+				Type:    testCheckType,
+				State:   health.HealthStateError,
+				Message: &messageInCaseOfError,
+				Params: map[string]interface{}{
+					"2": "Error #1 for key 2",
+				},
+			},
+			timeProvider: NewOrdinaryTimeProvider(),
+		},
+		{
+			name: "healthy when outside window",
+			keyErrorPairs: []keyErrorPair{
+				{key: "1", err: werror.Error("Error #1 for key 1")},
+			},
+			expectedCheck: whealth.HealthyHealthCheckResult(testCheckType),
+			timeProvider: &sliceTimeProvider{[]time.Time{
```

Comment:
Agreed, thanks for the suggestion. Added a configurable duration between submissions and evaluation, which is nice since it makes use of the default zero value and can be omitted except when needed